bento_lib/discovery/models/__init__.py

```diff
@@ -1,18 +1,21 @@
-from . import config, fields, overview, search
```

Comment:
I would create a new module in `bento_lib` called "provenance" for these models; it's not just for dataset, since we should have both project and dataset inheriting most of these common properties (which I should have made clearer in the original ticket.)